process/block/shardblock.go

```diff
@@ -377,7 +377,12 @@ func (sp *shardProcessor) checkMetaHdrFinality(header data.HeaderHandler) error
 	}
 
 	if nextBlocksVerified < sp.metaBlockFinality {
-		go sp.onRequestHeaderHandlerByNonce(lastVerifiedHdr.GetShardID(), lastVerifiedHdr.GetNonce()+1)
+		for nonce := lastVerifiedHdr.GetNonce(); nonce <= lastVerifiedHdr.GetNonce()+1; nonce++ {
+			go func(requestedNonce uint64) {
```

Comment:
same as for metachain. 2 calls are easier to understand than this for.

Reply:
done